File: src/p0001/java/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        var dict = new HashMap<Integer, Integer>();

        for (int j = 0; j < nums.length; j++) {
            var i = dict.get(target - nums[j]);
            if (i != null) {
                return new int[]{i, j};
            }

            dict.put(nums[j], j);
        }

        return new int[]{};
    }
}

File: src/p0001/rust/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut dict = HashMap::new();

        for (j, y) in nums.iter().enumerate() {
            if let Some(i) = dict.get(&(target - *y)) {
                return vec![*i, j as i32];
            }

            dict.insert(*y, j as i32);
        }

        return vec![];
    }
}
